src/main/java/co/cask/hydrator/salesforce/plugin/source/batch/SalesforceBatchSource.java

```diff
@@ -135,19 +134,20 @@ public void transform(KeyValue<NullWritable, CSVRecord> input,
         String fieldName = entry.getKey();
         String value = entry.getValue();
 
-        Schema.Field field = schema.getField(fieldName);
+        Schema.Field field = schema.getField(fieldName, true);
 
         if (field == null) {
           continue; // this field is not in schema
         }
 
-        builder.set(fieldName, convertValue(value, field));
+        builder.set(field.getName(), convertValue(value, field));
```

Comment:
why the change here?

Reply:
fields in salesforce are case-insensitive. And we saved them all lowercase but in this place we need to know the correct case field in CDAP schema has to be able to set the value via builder.